pkg/discovery/discover_test.go

```diff
@@ -0,0 +1,548 @@
+		Body: []byte(`[{}]`),
+	}
+
+	mc.reqs["/exchange/api/v2/assets/123/456/89"] = &api.Response{
+		Code: 400,
+		Body: nil,
+	}
+
+	asset := &anypoint.Asset{}
+
+	a := &anypoint.API{
+		ID:           456,
+		EndpointURI:  "google.com",
+		GroupID:      "123",
+		AssetID:      "456",
+		AssetVersion: "89",
+	}
+
+	_, err := agent.getServiceDetail(asset, a)
+	assert.NotNil(t, err)
+}
+
+func TestGetServiceDetailWhenExchangeAssetSpecFileIsEmpty(t *testing.T) {
+	agent, mc := buildAgentWithCustomMockAnypointClient()
+
+	mc.reqs["/apimanager/api/v1/organizations/333/environments/111/apis/456/policies"] = &api.Response{
+		Code: 200,
+		Body: []byte(`[{}]`),
+	}
+
+	mc.reqs["/exchange/api/v2/assets/123/456/89"] = &api.Response{
+		Code: 200,
+		Body: []byte("{\"Files\": [{}]}"),
+	}
+
+	asset := &anypoint.Asset{}
+
+	a := &anypoint.API{
+		ID:           456,
+		EndpointURI:  "google.com",
+		GroupID:      "123",
+		AssetID:      "456",
+		AssetVersion: "89",
+	}
+
+	sd, _ := agent.getServiceDetail(asset, a)
+	assert.Nil(t, sd)
+}
+
+func TestGetServiceDetailFailsWhenGetSpecFromExchangeFileFails(t *testing.T) {
+	agent, mc := buildAgentWithCustomMockAnypointClient()
+
+	mc.reqs["/apimanager/api/v1/organizations/333/environments/111/apis/456/policies"] = &api.Response{
+		Code: 200,
+		Body: []byte(`[{}]`),
+	}
+
+	mc.reqs["/exchange/api/v2/assets/123/456/89"] = &api.Response{
+		Code: 200,
+		Body: []byte("{\"Files\": [{\"Classifier\":\"oas\",\"Generated\":false, \"ExternalLink\": \"https://localhost/swagger.json\"}]}"),
+	}
+
+	mc.reqs["https://localhost/swagger.json"] = &api.Response{
+		Code: 400,
+		Body: nil,
+	}
+
+	asset := &anypoint.Asset{}
+
+	a := &anypoint.API{
+		ID:           456,
+		EndpointURI:  "google.com",
+		GroupID:      "123",
+		AssetID:      "456",
+		AssetVersion: "89",
+	}
+
+	_, err := agent.getServiceDetail(asset, a)
+	assert.NotNil(t, err)
+}
+
+func TestGetServiceDetailFailsWhenGetExchangeAssetIconFails(t *testing.T) {
+	agent, mc := buildAgentWithCustomMockAnypointClient()
+
+	mc.reqs["/apimanager/api/v1/organizations/333/environments/111/apis/456/policies"] = &api.Response{
+		Code: 200,
+		Body: []byte(`[{}]`),
+	}
+
+	mc.reqs["/exchange/api/v2/assets/123/456/89"] = &api.Response{
+		Code: 200,
+		Body: []byte("{\"Icon\": \"blahblah.com\", \"Files\": [{\"Classifier\":\"oas\",\"Generated\":false, \"ExternalLink\": \"https://localhost/swagger.json\"}]}"),
+	}
+
+	mc.reqs["https://localhost/swagger.json"] = &api.Response{
+		Code: 200,
+		Body: []byte("{\"basePath\":\"google.com\",\"host\":\"\",\"schemes\":[\"\"],\"swagger\":\"2.0\"}"),
+	}
+
+	mc.reqs["blahblah.com"] = &api.Response{
+		Code: 400,
+		Body: nil,
+	}
+
+	asset := &anypoint.Asset{}
+
+	a := &anypoint.API{
+		ID:           456,
+		EndpointURI:  "google.com",
+		GroupID:      "123",
+		AssetID:      "456",
+		AssetVersion: "89",
+	}
+
+	_, err := agent.getServiceDetail(asset, a)
+	assert.NotNil(t, err)
+}
+
+func TestShouldDiscoverAPIBasedOnTags(t *testing.T) {
```

Comment:
What happens in this test if an api has a tag that is included in the discover, and the do not discover arrrays?

Reply:
Good catch, it doesn't discover since the ignore tags are checked first..added this scenario to the test